src/protocol.rs

```diff
@@ -217,10 +218,15 @@ impl SignatureShare {
         if bool::from(s.is_high()) {
             s = s.negate();
         }
-        let r = self.r.ok_or(InternalError::NoChainedShares)?;
-
-        k256::ecdsa::Signature::from_scalars(r, s)
-            .map_err(|_| InternalError::SignatureInstantiationError)
+        let r = self.r.ok_or_else(|| {
+            error!("Tried to produce a signature without including shares");
+            InternalError::InternalInvariantFailed
+        })?;
+
+        k256::ecdsa::Signature::from_scalars(r, s).map_err(|_| {
+            error!("Unable to create ECDSA signature from provided r and s");
+            InternalError::InternalInvariantFailed
```

Comment:
todo: I think both of these are `ProtocolError`s, since they're functions of data from other parties.

Reply:
I think the first one is a result of forgetting to call `chain()` on your `SignatureShare`

Second one I think falls under the category "ultimately caused by bad inputs, but ones which should have been caught at an earlier check". What do you call that? I'm thinking `InternalInvariantFailed` is appropriate since it's really our oops if this gets hit, even with malicious inputs.

Reply:
Okay, that's fair. Let's do IIF then.